parsec/api/protocol/handshake.py

```diff
@@ -461,8 +476,16 @@ def __init__(
     def process_challenge_req(self, req: bytes) -> bytes:
         self.load_challenge_req(req)
         challenge = self.challenge_data["challenge"]
+
         assert isinstance(challenge, bytes)
-        answer = self.user_signkey.sign(challenge)
+
+        # TO-DO remove the else for the next release
+        if self.backend_api_version.version >= 2 and self.backend_api_version.revision >= 5:
+            # TO-DO Need to use "BaseSignedData" ?
```

Comment:
This seems fine enough to me ;-)

The only drawback I see is the fact that sign + dump must be done by the caller instead of behind provided as a clear api. But given this is only used in a single place (and this is not going to change), it is not a big issue.

If you want to avoid this, you could create your own class `AnswerSerializer` inheriting `MsgpackSerializer` which would define `dump_and_sign/verify_and_load` methods.